src/app/home/home.page.ts

```diff
@@ -137,26 +137,32 @@ export class HomePage {
     this.isLoading = true;
 
     const cachedData = localStorage.getItem(`population_${item.iso3}`);
-    if (cachedData) {
-      item.population = JSON.parse(cachedData);
-      this.updatedLoadingState(item);
-    } else {
-      this.countryService.getSingleCountryPopulation(item.iso3).subscribe({
-        next: (result) => {
-          if (!result.error) {
-            item.population = result.data.populationCounts?.slice(-1)[0];
-            localStorage.setItem(`population_${item.iso3}`, JSON.stringify(result.data.populationCounts?.slice(-1)[0]));
-            this.updatedLoadingState(item);
-          }
-        },
-        error: () => {
+    cachedData ? this.handleCachedPopulation(item, cachedData) : this.fetchPopulation(item);
+  }
+  private handleCachedPopulation(item: Country, cachedData: string): void {
+    item.population = JSON.parse(cachedData);
+    this.updatedLoadingState(item);
+  }
+  private fetchPopulation(item: Country): void {
+    this.countryService.getSingleCountryPopulation(item.iso3).subscribe({
```

Comment:
1-close stream use $destroy
2- prevent duplicate code of  item.population = { value: 0 } move it to complete event in subscriber